src/main/java/org/cardanofoundation/cfexploreraggregator/txcount/model/repository/AddressTxCountRepository.java

```diff
@@ -12,4 +14,9 @@ public interface AddressTxCountRepository extends JpaRepository<AddressTxCountEn
 
     void deleteBySlotGreaterThan(long slot);
 
+    @Query(nativeQuery = true, value = """
+        DELETE FROM address_tx_count WHERE slot < :slot
+        AND address IN (SELECT address FROM address_tx_count GROUP BY address HAVING COUNT(*) > 1)
```

Comment:
is there any chance `:slot` value is higher than the most recent address tx count and hence delete also the most recent value and not only the old one?